### DISCUSSION TITLE
unexpected result when using gf (goto file in selection)

### DISCUSSION THREAD

**Discussion by sn0n** [helix-editor/helix#9482] [General]
### Summary

using the goto file in a .md file, the link will bug out the buffers path/opened file.
(I did learn about `gd` as a solution, but file should work with or without a space)

### Reproduction Steps

 

I tried this:

1. use 'gf' on a file when the file is a markdown link.
ex: `[title](./file.md)`

I expected this to happen:
to buffer `./file.md`
Instead, this happened:
it buffered `[title](./file.md)`

there is also other unexpected behaviour when the filename has a space in it, `[some link](somewhere.md)` 

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2024-01-29T05:25:34.594 helix_lsp::transport [ERROR] marksman err <- "[05:25:34 INF] <LSP Entry> Starting Marksman LSP server: {}\n"
2024-01-29T05:25:57.129 helix_view::editor [ERROR] editor error: Register [@] empty
```

</details>

### Platform

Linux

### Terminal Emulator

alacritty

### Installation Method

extra (arch linux)

### Helix Version

23.10-1

**Comment by sn0n** [comment]
I do want to mention that yes, I do have marksman installed, but the same bug was happening before it was installed, I only found marksman as a result of some googling trying to fix the issue.

**Comment by woojiq** [comment]
Iirc, with marksman you can use `goto definition` instead of `goto file` to open source file.
And what about `gf` in general. It's hard to find out what users want to open without selection. That's why you need to select a file and then press `gf` or it might be broken otherwise.

**Reply by sn0n** [reply]
I now have marksman installed and I'm using `gd` going forward, but `gf` does not work either way.. with or without marksman.

**Reply by woojiq** [reply]
It works for me. Are you sure you visually selected file path before `gf`?
![image](https://github.com/helix-editor/helix/assets/122799969/34763bd2-f797-48ed-8417-bdbd263d20cf)

**Reply by sn0n** [reply]
Ohhhhh, sorry, I wasn't selecting, Because it does work without selecting otherwise. 
 My mistake, apologies.

**Reply by woojiq** [reply]
> Because it does work without selecting otherwise.

This is because it really does work if you have simple cases where you don't surround the file path with other characters. Without selection it tries to expand file path to the left and right until space.
